src/shiver/models/histogram.py

```diff
@@ -175,16 +176,31 @@ def generate_mde(self, config_dict: dict) -> str:
 
     def clone(self, ws_name, ws_clone_name):
         """Clone the workspace"""
-
         CloneMDWorkspace(InputWorkspace=ws_name, OutputWorkspace=ws_clone_name)
 
     def scale(self, ws_name_in, ws_name_out, scale_factor):
         """Scale the workspace"""
-
         ws_scalefactor = CreateSingleValuedWorkspace(OutputWorkspace="scalefactor", DataValue=scale_factor)
         MultiplyMD(LHSWorkspace=ws_name_in, RHSWorkspace=ws_scalefactor, OutputWorkspace=ws_name_out)
         DeleteWorkspace(ws_scalefactor)
 
+        # update the sample log
+        # if previous scale factor applied: multiply by this scale factor
+        # else add a new entry with this scale factor
+        run = mtd[ws_name_out].getExperimentInfo(0).run()
+        if "MDScale" in run.keys():
+            AddSampleLog(
```

Comment:
Please, add a test to cover the scenario that a sample log with "MDScale" key already exists.

Reply:
Done